xds/internal/balancer/outlierdetection/e2e_test/outlierdetection_test.go

```diff
@@ -367,3 +383,63 @@ func (s) TestNoopConfiguration(t *testing.T) {
 		t.Fatalf("error in expected round robin: %v", err)
 	}
 }
+
+// Test verifies that outlier detection doesn't eject subchannels created by
+// the new pickfirst balancer when pickfirst is a top-level policy.
+func (s) TestPickFirstHealthListenerDisabled(t *testing.T) {
+	backend := &stubserver.StubServer{
+		EmptyCallF: func(ctx context.Context, in *testpb.Empty) (*testpb.Empty, error) {
+			return nil, errors.New("some error")
+		},
+	}
+	if err := backend.StartServer(); err != nil {
+		t.Fatalf("Failed to start backend: %v", err)
+	}
+	defer backend.Stop()
+	t.Logf("Started bad TestService backend at: %q", backend.Address)
+
+	countingODServiceConfigJSON := fmt.Sprintf(`
+	{
+	  "loadBalancingConfig": [
+		{
+		  "outlier_detection_experimental": {
+			"interval": "0.025s",
+			"baseEjectionTime": "0.100s",
+			"maxEjectionTime": "300s",
+			"failurePercentageEjection": {
+				"threshold": 50,
+				"enforcementPercentage": 100,
+				"minimumHosts": 0,
+				"requestVolume": 2
+			},
+			"childPolicy": [{"%s": {}}]
+		  }
+		}
+	  ]
+	}`, pickfirstleaf.Name)
+	sc := internal.ParseServiceConfig.(func(string) *serviceconfig.ParseResult)(countingODServiceConfigJSON)
+
+	mr := manual.NewBuilderWithScheme("od-e2e")
+
+	mr.InitialState(resolver.State{
+		Addresses:     []resolver.Address{{Addr: backend.Address}},
+		ServiceConfig: sc,
+	})
+	cc, err := grpc.NewClient(mr.Scheme()+":///", grpc.WithResolvers(mr), grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		t.Fatalf("grpc.NewClient() failed: %v", err)
+	}
+	defer cc.Close()
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+	testServiceClient := testgrpc.NewTestServiceClient(cc)
+
+	// Failing request should not cause ejection.
```

Comment:
Is this because `pickfirstleaf.EnableHealthListener` is not invoked by any petiole policy, so pickfirst does not register a health listener? And the subchannel wrapper in OD has `healthListenerEnabled` set to `true` because the subchannel is managed by pickfirst, and therefore when OD processes an ejection update, and sees that health listener is enabled, but not listener function is registered, it drops ejection updates on the floor?

If that is the case, can we have that clarified in the top-level test comment? Thanks.